pkg/router/template/template_helper_test.go

```diff
@@ -805,3 +807,71 @@ func TestClipHAProxyTimeoutValue(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateHAProxyWhiteListFile(t *testing.T) {
+	workDir, err := ioutil.TempDir("", "haproxy")
+	if err != nil {
+		t.Fatal("Unable to create a temporary working directory")
+	}
+	defer os.RemoveAll(workDir)
+
+	err = os.MkdirAll(path.Join(workDir, whitelistDir), 0740)
+	if err != nil {
+		t.Fatal("Unable to create the whitelist directory")
+	}
+
+	testCases := []struct {
+		name              string
+		workDir           string
+		id                ServiceAliasConfigKey
+		expectedWhiteList []string
+		failureExpected   bool
+	}{
+		{
+			name:    "Nominal",
+			workDir: workDir,
+			id:      ServiceAliasConfigKey("test1"),
+			expectedWhiteList: []string{
+				"192.168.0.1",
+				"192.168.0.2",
+				"192.168.0.3",
+			},
+		},
+		{
+			name:    "Nominal failure",
+			workDir: workDir + "-notexisting",
+			id:      ServiceAliasConfigKey("test2"),
+			expectedWhiteList: []string{
+				"192.168.0.1",
+				"192.168.0.2",
+				"192.168.0.3",
+			},
+			failureExpected: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			file := generateHAProxyWhiteListFile(tc.workDir, tc.id, strings.Join(tc.expectedWhiteList, " "))
+			if tc.failureExpected {
+				if file != "" {
+					t.Fatal("Failure expected but didn't happen")
+				}
+				return
+			} else {
+				if file == "" {
+					t.Fatal("Not expected failure happened")
+				}
+			}
+
+			contents, err := ioutil.ReadFile(file)
+			if err != nil {
+				t.Fatalf("Unable to read from the generated file: %q", err)
```

Comment:
Is there a reason to use `%q` here?  
```suggestion
				t.Fatalf("Unable to read from the generated file: %v", err)
```

Reply:
No, not much, I thought it would look a bit more clear, but I think you are right - `%v` would do this thing.